Mu2eUtilities/src/ParametricFit.cc

```diff
@@ -0,0 +1,277 @@
+//Author: S Middleton
+//Purpose: Stores all the functions associated with building and interpretting parametric line equations for the purpose of cosmic track based alignment. This includes calculting track axes, hit errors and re-orientating those errors, track residuals are also calcuated here.
+
+#include "Mu2eUtilities/inc/ParametricFit.hh"
+#include "RecoDataProducts/inc/CosmicTrack.hh"
+
+//ROOT
+#include "TMath.h"
+
+using namespace std;
+using namespace mu2e;
+
+
+namespace ParametricFit{
+
+	double GettMin(XYZVec& point, XYZVec& starting_point, XYZVec& end_point){
+            double tMin = -(starting_point-point).Dot(end_point-starting_point) /((end_point-starting_point).Mag2());
+	    return tMin;
+
+       }
+
+	XYZVec PointToLineCA(XYZVec& point, XYZVec& starting_point, XYZVec& end_point){
```

Comment:
This function should be its own class, PointLinePCA_XYZ to follow the existing naming convention in Mu2eUtilities.